File: 1684 count-the-number-of-consistent-strings/solution.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        vector<bool> set(26,0);
        for(char c:allowed) set[c-'a'] = true;
        int ans = 0;
        for(string s:words) {
            bool is = true;
            for(char c:s) {
                if(!set[c-'a']) {
                    is = false;
                    break;
                }
            }
            if(is) ans++;
        }
        return ans;
    }
};

File: 1684 count-the-number-of-consistent-strings/solution.rs
impl Solution {
    pub fn count_consistent_strings(allowed: String, words: Vec<String>) -> i32 {
        let mut set = vec![false;126];
        for c in allowed.chars() {
            set[c as usize] = true;
        }
        let mut ans = 0;
        for w in words {
            let mut is = true;
            for c in w.chars() {
                if !set[c as usize] {
                    is = false;
                    break;
                }
            }
            if is {
                ans+=1;
            }
        }
        ans
    }
}
